cxx/structureTranslationModule/translator/NrelFromQuasybinaryNodeTranslator.cpp

```diff
@@ -19,48 +19,57 @@ NrelFromQuasybinaryNodeTranslator::NrelFromQuasybinaryNodeTranslator(ScMemoryCon
+      ScType::EdgeAccessVarPosPerm,
       ScType::NodeVarNoRole >> TranslationConstants::NREL_ALIAS);
-  context->HelperSmartSearchTemplate(scTemplate,
-  [&](ScTemplateResultItem const & searchResult)
-  {
-    node = searchResult[TranslationConstants::NODE_ALIAS];
-    tupleNode = searchResult[TranslationConstants::TUPLE_ALIAS];
-    nrelNode = searchResult[TranslationConstants::NREL_ALIAS];
-    std::string const & nodeMainIdtf = utils::CommonUtils::getMainIdtf(context, node, {TranslationKeynodes::lang_en});
-    if (nodeMainIdtf.empty())
-      return ScTemplateSearchRequest::CONTINUE;
-    std::string const & nrelMainIdtf = utils::CommonUtils::getMainIdtf(context, nrelNode, {TranslationKeynodes::lang_en});
-    if (nrelMainIdtf.empty())
-      return ScTemplateSearchRequest::CONTINUE;
-
-    auto const & tupleNodeIterator = context->Iterator3(tupleNode, ScType::EdgeAccessConstPosPerm, ScType::NodeConst);
+  context->HelperSmartSearchTemplate(
+      scTemplate,
+      [&](ScTemplateResultItem const & searchResult)
+      {
+        node = searchResult[TranslationConstants::NODE_ALIAS];
+        tupleNode = searchResult[TranslationConstants::TUPLE_ALIAS];
+        nrelNode = searchResult[TranslationConstants::NREL_ALIAS];
+        std::string const & nodeMainIdtf =
+            utils::CommonUtils::getMainIdtf(context, node, {scAgentsCommon::CoreKeynodes::lang_ru});
+        if (nodeMainIdtf.empty())
+          return ScTemplateSearchRequest::CONTINUE;
+        std::string const & nrelMainIdtf =
+            utils::CommonUtils::getMainIdtf(context, nrelNode, {scAgentsCommon::CoreKeynodes::lang_ru});
+        if (nrelMainIdtf.empty())
+          return ScTemplateSearchRequest::CONTINUE;
 
-    while (tupleNodeIterator->Next())
-    {
-      ScAddr const & tupleNode = tupleNodeIterator->Get(2);
-      std::string const & tupleNodeMainIdtf = utils::CommonUtils::getMainIdtf(context, tupleNode, {TranslationKeynodes::lang_en});
-      if (tupleNodeMainIdtf.empty())
-        continue;
-      translations << nodeMainIdtf << " " << nrelMainIdtf << " " << tupleNodeMainIdtf << ". ";
-    }
-    return ScTemplateSearchRequest::CONTINUE;
-  },
-  [&](ScAddr const & element)
-  {
-    return isInStructure(structAddr, element);
-  });
+        auto const & tupleNodeIterator =
+            context->Iterator3(tupleNode, ScType::EdgeAccessConstPosPerm, ScType::NodeConst);
 
+        while (tupleNodeIterator->Next())
+        {
+          ScAddr const & tupleNode = tupleNodeIterator->Get(2);
+          std::string const & tupleNodeMainIdtf =
+              utils::CommonUtils::getMainIdtf(context, tupleNode, {scAgentsCommon::CoreKeynodes::lang_ru});
+          if (tupleNodeMainIdtf.empty())
+            continue;
+          translation << tupleNodeMainIdtf << ", ";
+        }
+        if (!(translation.tellp() == 0))
+          translations << nodeMainIdtf << " " << nrelMainIdtf << " " << translation.str();
+        translation.str("");
```

Comment:
You already checked if idtf empty. What is it for?

Reply:
It is to make the string empty, so already written text doesn't show up in future parts of translations